docs/guides/color.md

```diff
@@ -0,0 +1,290 @@
+For example, a 2D image of floats between 0 and 1 can be mapped to colors.
+
+```python
+In [6]: image.colormap = 'blue'
+```
+
+Internally, the [`ensure_colormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/colormap_utils.py#L492)
+function is used to coerce colormap input values to an instance of napari's `Colormap`.
+
+```python
+In [7]: image.colormap
+Out[7]:
+Colormap(
+    colors=np.array([0., 0., 0., 1], [0., 0., 1., 1]]),
+    name='blue',
+    interpolation=<ColormapInterpolationMode.LINEAR: 'linear'>,
+    controls=array([0., 1.]),
+)
+```
+
+In this case, values of 0 are mapped to black (`[0, 0, 0, 1]`),
+values of 1 are mapped to blue (`[0, 0, 1, 1]`),
+and values in `(0, 1)` are mapped to colors that are linearly interpolated
+between black and blue.
+
+A few types of input are supported.
+The easiest to use is the colormap name as above.
+```python
+image.colormap = 'blue'
+```
+
+By default colormap names come from one of the following sources.
+
+- Some of [matplotlib's colormap names](https://matplotlib.org/stable/tutorials/colors/colormaps.html).
+- Standard optical primary and secondary colors: `'red', 'green', 'blue', 'yellow', 'magenta', 'cyan'`.
+- BOP (blue, orange, purple) colormap names: `'bop blue', 'bop orange', 'bop purple'`.
+
+More specifically, the available colormap names can be listed at any time.
+
+```python
+In [8]: list(napari.utils.colormaps.AVAILABLE_COLORMAPS)
+Out[8]:
+['PiYG',
+ 'blue',
+ 'bop blue',
+ ...,
+ 'viridis',
+ 'yellow']
+```
+
+If there is a need for a custom colormap that is not available by default,
+pass an instance of a napari `Colormap`
+
+
+```python
+image.colormap = Colormap(
+    colors=[[0, 0, 0, 1], [0, 0, 1, 1]],
+    name='blue',
+    interpolation='linear',
+    controls=[0, 1],
+)
+```
+
+which will be added to `AVAILABLE_COLORMAPS` after it is successfully created.
+Equivalently, pass a `dict` representation of the same colormap.
+
+```python
+image.colormap = {
+    'colors': [[0, 0, 0, 1], [0, 0, 1, 1]],
+    'name': 'blue',
+    'interpolation': 'linear',
+    'controls': [0, 1],
+}
+```
+
+Any single color value that is not a string can also be used
+
+```python
+image.colormap = (0, 0, 1)
+```
+
+in which case a colormap from black to the given color will be generated.
+In general, the color value cannot be given as a name string because strings
+are reserved to refer to existing colormap names.
+
+
+### Categorical maps
+
+The [`CategoricalColormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/categorical_colormap.py#L13)
+class defines a map from a categorical or discrete value to a color.
+It is defined by two components.
+
+1. A dictionary that maps from values to colors.
+2. A cycle of fallback colors that used when a value is not found in the dictionary.
+
+For example, a points layer's face color could be mapped from an optional cell type
```

Comment:
I think the setup for this example should introduce the layer data and property values. Without, I think this would be a bit confusing for people that aren't already familiar with coloring points with properties.

This would be an example of something that might be better in a "how-to" document than an "explanatory" document.

Reply:
That's a good point. My goal was to mainly focus on explanation here, partly because we already have some how-to-ish docs in the layer tutorials.

Maybe it would make more sense to discuss the `map` method of `Colormap` and `CategoricalColormap` instead of the points example to better describe how these things work, rather than how they are used in napari?

Reply:
I agree that it's best to focus on the explanation rather than the implementation. In light of that, I think your idea to discuss `map` from `Colormap` and `CategoricalColormap` is a good alternative!